galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/ITsq.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.cicsts;
+
+import javax.validation.constraints.NotNull;
+public interface ITsq {
+
+      /**
+       * Chect the existence of a TSQ. 
+       * @throws TsqException if there is a problem in checking the TSQ existence
+       * @return boolean based on if TSQ is existing or not
+       */
+      public boolean exists() throws TsqException;  
+
+      /**
+       * Check if a TSQ is recoverable. 
+       * @throws TsqException if there is a problem in checking if the TSQ is recoverable or not
+       * @return boolean based on if TSQ is recoverable or not
+       */      
+      public boolean isRecoverable() throws TsqException;
+
+      /**
+       * Read Data from TSQ based on item number.
+       * @param item Item number of the TSQ to be read
+       * @return Data read from TSQ as String 
+       * @throws TsqException if there is a problem in reading from the TSQ
+       */    
+      public String readQueue(@NotNull int item) throws TsqException;
```

Comment:
ints can't be null, so the annotation is bad.
But if the result could be a null, it should be Optional(String) returned ?

Reply:
Removed @NotNull.